mkosi/__init__.py

```diff
@@ -2255,3 +2270,27 @@ def run_verb(args: MkosiArgs, presets: Sequence[MkosiConfig]) -> None:
 
             if args.verb == Verb.serve:
                 run_serve(last)
+
+
+def normalize_mtime(source_date_epoch: Optional[int], root: Path, target: Optional[Path] = None) -> None:
+    if source_date_epoch is None:
+        return
+
+    # If no target given, normalize the mtime recursively for the entire tree, starting at root.
+    if not target:
+        with complete_step("Setting mtime to SOURCE_DATE_EPOCH"):
+            os.utime(root, (source_date_epoch, source_date_epoch), follow_symlinks=False)
+            for p in root.rglob("*"):
```

Comment:
The comment says recursively, but the single `*` will get you down at most one directory level, do you mean
```suggestion
            for p in root.rglob("**/*"):
```

Reply:
This is using [`rglob`](https://docs.python.org/3/library/pathlib.html#pathlib.Path.rglob), not glob, so it works as described.